### PR TITLE
Fix the unstable resource manager testcase

### PR DESCRIPTION
**Pull request by SimFG**
issue: #28030
unauthenticated retry issue: [1629](https://github.com/milvus-io/pymilvus/issues/1629)
/kind improvement

### CODE REVIEW

File: pkg/util/resource/resource_manager_test.go [hunk 1]
```diff
@@ -63,7 +63,9 @@ func TestResourceManager(t *testing.T) {
 		assert.Equal(t, "foo2", res.Get())
 		res = manager.Delete("test", "foo")
 		assert.Equal(t, "foo2", res.Get())
-		time.Sleep(time.Second)
+		for manager.Delete("test", "foo") != nil {
+			time.Sleep(500 * time.Millisecond)
+		}
```
Comment (congqixia) [inline pkg/util/resource/resource_manager_test.go:68]: NIT: we could use testify `Eventually` function to perform this periodic check with timeout 
```suggestion
		assert.Eventually(t, func() bool {
			return manager.Delete("test", "foo") != nil
		}, time.Second*5, time.Millisecond*100)
```

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**